docs/docs/guides/04-fe-coding-guidelines.mdx

```diff
@@ -0,0 +1,596 @@
+
+export default MyButton;
+
+// form.tsx
+import Button from './button';
+
+// NAMED EXPORT
+export const Form = () => <Button type="button">Submit</Button>;
+```
+
+Above is a simple component for illustration purposes, but you'd be likely to use the `MyButton`
+component in _a lot_ of places, so what happens if you want to replace all instances with a new v2
+Button? If the naming is not as consistent, it's typically a lot more frustrating & can lead to
+bugs.
+
+### Utilities / Helpers
+
+When you need a utility function ask yourself a question – do I need this _anywhere_ else? If you
+don't (which is fairly typical) include it in the file you're using it. As a guide, we expect
+anything exported to be unit-tested and if a utility or hook is unique and bespoke to the component
+you're creating then it's advised to test that code in the context of said component, this is
+covered in the Testing guides below.
+
+If you do need to use the utility elsewhere then follow the same practices shown in
+[Application Structure](#application-structure) keep it as close to where it's used as possible.
+Avoid re-exporting them from a generic `index` file. However, you might find it useful to group them
+under a generic file name such as `strings.ts` which would include all the `string` utils you have.
+
+### Components & Hooks
+
+With components and hooks, to improve code discoverability, opt for this pattern instead:
+
+```shell
+useContentManager/
+├─ index.ts
+├─ useContentManager.ts
+```
+
+### Memoization
+
+Premature optimisation is the root of all evil. Performance of the application _is_ important
+however it should be measurable before we attempt to solve the issue as these
+functions _are not free._ Take this example:
+
+```jsx
+const FormButton = () => {
+  const handleSubmit = useCallback((entity) => {
+    post('/entity/create', {
+      entity,
+    });
+  }, []);
+
+  return <Button onClick={handleSubmit}>Create Entity</Button>;
+};
+
+const Button = ({ onClick, children }) => <button onClick={onClick}>{children}</button>;
+```
+
+The component we're passing to is _very_ simple & `useCallback` is unnecessary here. Because we do
+not need the function identity to be stable. If we were to use the callback in a `useEffect` hook
+then yes, we would want the function to be stable otherwise every render would cause the `useEffect`
+hook to run which could be problematic.
+
+So what about `useMemo`. Same thing, it's for heavy computational processes e.g. not this:
+
+```jsx
+const Component = ({ entities = [] }) => {
+  const entityLength = useMemo(() => entities.length, [entities]);
+  //...
+};
+```
+
+The above function isn't expensive but more importantly we don't know what impact it has without
+measuring it. Therefore, avoid using `useCallback` and `useMemo` unless you _know_ you need it.
+
+## Dependencies
+
+Generally speaking, always opt to use native-browser APIs where possible over library code, the
```

Comment:
Does this mean we should avoid using lodash when possible? Even though the code readability might be better?

Reply:
Ok I kept reading and it's clear. Avoid lodash